Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1140,13 +1140,19 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
             return
         }
 
+        var deferredEventEmittingBlock: DeferredBlock? = nil
+        defer { deferredEventEmittingBlock?() }
+
         nsConfig.nsLock.writeLock()
         defer { nsConfig.nsLock.unlock() }
 
         nsConfig[documentId] = nil
         try self.localCollection(for: namespace,
                                  withType: Document.self).deleteOne(["_id": documentId])
-        triggerListening(to: namespace)
+
+        deferredEventEmittingBlock = {
+            self.triggerListening(to: namespace)
```

Comment:
this logic is not conditional on anything -- why not just defer this block directly? rather than defer the reference above then change the value

Reply:
There is the unlikely possibility that `deleteOne` might fail (in which case we have a larger problem), so we don't necessarily want to trigger listening to the namespace in that case